internal/driver/fetch.go

```diff
@@ -294,7 +294,7 @@ func setTmpDir(ui plugin.UI) (string, error) {
 	dirs = append(dirs, os.TempDir())
 	for _, tmpDir := range dirs {
 		if err := os.MkdirAll(tmpDir, 0755); err != nil {
-			ui.PrintErr("Could not use temp dir ", tmpDir, ": ", err.Error())
+			ui.Print("Could not use temp dir ", tmpDir, ": ", err.Error())
```

Comment:
I don't think it's the right way to fix the single test on a single platform, as it changes the behavior globally.

You can conditionally (for the problematic environment only) add a pattern to ignore in the configuration of the TestUI in the test, https://github.com/google/pprof/blob/79c4198ef7bd1069f8f56501fc05f0f1d2c33d8a/internal/driver/fetch_test.go#L416 which should be sufficient.

I also wonder if the
```
proftest.go:114: Could not use temp dir /private/var/mobile/Containers/Data/Application/2B578AA2-4677-47C6-8665-DDB9509D5C92/pprof: mkdir /private/var/mobile/Containers/Data/Application/2B578AA2-4677-47C6-8665-DDB9509D5C92/pprof: operation not permitted
```
message from the test represents a real issue with temporary directory location on iOS. That can be handled separately though.

Reply:
setTmpDir tries $HOME/pprof first before os.TempDir. On iOS $HOME is the base directory of the installed app and isn't writable so I assumed the "Could not use temp dir" should be a warning, not an error.

However, I changed the PrintErr back and made your suggested change to the ignore pattern.